bee-message/src/output/foundry_id.rs

```diff
@@ -15,19 +15,16 @@ string_serde_impl!(FoundryId);
 
 impl FoundryId {
     /// Builds a new [`FoundryId`] from its components.
-    pub fn build(alias_id: AliasId, serial_number: u32, token_scheme: TokenScheme) -> Self {
-        Self(
-            [AliasAddress::KIND]
-                .iter()
-                .chain(alias_id.as_ref())
-                .chain(&serial_number.to_le_bytes())
-                .chain(&[token_scheme as u8])
-                .copied()
-                .collect::<Vec<u8>>()
-                .try_into()
-                // SAFETY: the lengths are known.
-                .unwrap(),
-        )
+    pub fn build(alias_address: &AliasAddress, serial_number: u32, token_scheme: TokenScheme) -> Self {
+        let mut bytes = [0u8; FoundryId::LENGTH];
+        let mut packer = SlicePacker::new(&mut bytes);
+
+        // SAFETY: packing to an array of the correct length can't fail.
```

Comment:
Might come across as too nit-picky, but the methods below aren't unsafe, and I think SAFETY tag is only used specifically for calls to unsafe code (although I am not 100% sure).

Reply:
I changed them to PANIC and opened an issue to also do it for the whole repo https://github.com/iotaledger/bee/issues/1270